apps/user_status/lib/Service/StatusService.php

```diff
@@ -284,14 +286,23 @@ public function setUserStatus(string $userId,
 			}
 		}
 
+		$now = $this->timeFactory->getTime();
+
+		$clearAt = $extraParams['clearAt'] ?? null;
+		if ($clearAt instanceof \DateTime) {
+			$clearAt = $clearAt->getTimestamp();
+		} elseif (!is_int($clearAt) || $clearAt < $now) {
```

Comment:
Should that be in else?
What if $clearAt is a datetime which is before now?

Reply:
So it should be like:
```suggestion
		}
		if (!is_int($clearAt) || $clearAt < $now) {
```

Reply:
Oh right, sorry.